### PR TITLE
Haskell

### PR DESCRIPTION
**Pull request by noFrostoo**

### CODE REVIEW

File: haskell/adventure.hs [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+\Towarzyszy mu jedynie deszcz i jego wierny koń Piwonia.\
+\Pewnego popołudnia dociera do Świętego Dębu.\
+\Niestety te sakramentalne miejsce zostało zbrukane ludzką krwią."
+describeLocation "WieśGrobla" = "Gerwant dociera do Wsi Grobla."
+describeLocation "DomSołtysa" = "Zabójca potworów wchodzi do domu sołtysa"
+describeLocation "Karczma" = "Gerwant wchodzi do Karczmy"
+describeLocation "PolanaKołoChatyDrwali" = "Zabójca potworów znajduje się na polanie wokoło Chaty Drwali."
+describeLocation "ChataDrwali" = "Łowca potworów wchodzi do Chaty Drwali"
+describeLocation "Puszcza" = "Mistrz Gerwant wkracza w samo serce puszczy."
+describeLocation "Karczma" = "Gerwant z Riviery wchodzi do pieczary. \
```
Comment (tejotemku) [inline haskell/adventure.hs]: Wrong location name, should be "Pieczara"

File: haskell/adventure.hs [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+      put $ 
+        game { itemLocations = Map.insert "WieśGrobla"  ( Set.insert "GnomskiGwyhyr" items) (itemLocations game) }
+
+advanceDay :: Monad m => StateT GameState m TimeOfDay
+advanceDay = do
+  game <- get
+  if dayTurnCounter game == 11
+    then do
+      let newTimeOfDay = if timeOfDay game == Day then Night else Day
+      put $ game {dayTurnCounter = 0, timeOfDay = newTimeOfDay}
```
Comment (tejotemku) [inline haskell/adventure.hs]: Consider printing that the time of day changes

File: haskell/adventure.hs [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+            }
+        when (gerwantHpAfter <= 0) (put initialState)
+        return $ "Gerwant atakuje za " ++ show gerwantAttack ++ " pkt.\n"
+          ++ if enemyHpAfter <= 0
+            then "Potwór pada martwy od ciosu."
+            else "Przeciwnik atakuje za "
+              ++ show enemyAttack
+              ++ " pkt.\n"
+              ++ if gerwantHpAfter <= 0
+                then "Wiedźmin pada od ciosu potwora. Gerwant umarł, a cała płeć żeńska na kontynencie uroniła łzę.\nGra zaczyna się od początku."
```
Comment (tejotemku) [inline haskell/adventure.hs:298]: sadge

File: haskell/adventure.hs [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+        then lift $ putStrLn "Gerwant pije piwo. Żywotność zostaje przywrócona."
+        else lift $ putStrLn "Nie masz wystarczająco złota."
+    ("przygotuj" : "się" : "do" : "walki" : args) -> checkNumArgs 0 args do
+      game <- get
+      if location game == "Pieczara"
+        then do
+          lift $ putStrLn "Do walki z jakim potworem musi się przygotować Gerwant z Riviery?"
+          monster <- lift readCommand
+          case monster of
+            ["leszy"] -> do
```
Comment (tejotemku) [inline haskell/adventure.hs:446]: What about first letter uppercase situation?

File: haskell/adventure.hs [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+    "polecenia              -- by wyświetlić listę poleceń.",
+    "idź <kierunek>         -- by pójść w danym kierunku.",
+    "podnieś <przedmiot>    -- by podnieść przedmiot.",
+    "upuść <przedmiot>      -- by upuścić przedmiot.",
+    "ekwipunek              -- by sprawdzić ekwipunek.",
+    "obejrzyj <przedmiot>   -- by obejrzeć przedmiot.",
+    "rozmawiaj <npc>        -- by porozmawiać z NPC.",
+    "spytaj <npc> <coś>     -- by spytać NPC o coś.",
+    "rozejrzyj się          -- by dowiedzieć się, gdzie jesteś.",
+    "przygotuj się do walki -- by odgadnąć rodzaj potwora i zwiększyć swoje szanse na wygraną.",
```
Comment (tejotemku) [inline haskell/adventure.hs]: This should be hidden and revealed when "Gerwant" steps into the "Pieczara" location

File: haskell/adventure.hs [hunk 6]
```diff
@@ -0,0 +1,426 @@
...
+\Niestety te sakramentalne miejsce zostało zbrukane ludzką krwią."
+describeLocation "WieśGrobla" = "Gerwant dociera do Wsi Grobla."
+describeLocation "DomSołtysa" = "Zabójca potworów wchodzi do domu sołtysa"
+describeLocation "Karczma" = "Gerwant wchodzi do Karczmy"
+describeLocation "PolanaKołoChatyDrwali" = "Zabójca potworów znajduje się na polanie wokoło Chaty Drwali."
+describeLocation "ChataDrwali" = "Łowca potworów wchodzi do Chaty Drwali"
+describeLocation "Puszcza" = "Mistrz Gerwant wkracza w samo serce puszczy."
+describeLocation "Karczma" = "Gerwant z Riviery wchodzi do pieczary. \
+\Gerwant wnioskuje, że natrafił na leże potwora. \
+\Monstrum nie ma w jaskini, więc może na niego zaczekać (komenda przygotujSięDoWalki). \
```
Comment (tejotemku) [inline haskell/adventure.hs]: since command is "przygotuj się do walki" this should also follow that format

File: haskell/adventure.hs [hunk 7]
```diff
@@ -0,0 +1,426 @@
...
+describeItem "Mieszek" = "Mieszek pełen złota. W środku znajduje się 10 monet. Wystarczy na piwo."
+
+describeItem "Statua" = " Analiza Gerwanta z Riviery: \"Medalion Drży. To magiczny totem. Podobny totem widziałem pod Świętym dębem\". Wiedźmin postanawia przeczytać bestiariusz. W wiedźmińskim almanachu Gerwant znajduje informację, że magiczne totemy to znak rozpoznawczny terytorium potwora zwanego jako *Leszy*."
+describeItem _ = "Nie ma takiego przedmiotu."
+
+describeLocation :: String -> String
+describeLocation "ŚwiętyDąb" = "Gerwant z Riviery, szkoły nosacza, podróżuje szlakami Królestw Północy już wiele dni.\
+\Towarzyszy mu jedynie deszcz i jego wierny koń Piwonia.\
+\Pewnego popołudnia dociera do Świętego Dębu.\
+\Niestety te sakramentalne miejsce zostało zbrukane ludzką krwią."
```
Comment (tejotemku) [inline haskell/adventure.hs]: This is a game introduction text, it does not fit the location description. Also if you start the game you will get ` 
"Gerwant z Riviery, szkoły nosacza, podróżuje szlakami Królestw Północy już wiele dni.\
\Towarzyszy mu jedynie deszcz i jego wierny koń Piwonia" ` text two times that way. Lets change it to ` "Gerwant dociera do Świętego Dębu.\
\Niestety te sakramentalne miejsce zostało zbrukane ludzką krwią." `

File: haskell/adventure.hs (hunk 5 above)
Comment (noFrostoo) [reply haskell/adventure.hs]: Removed from help, will not add it when revleaed

Comment (tejotemku) [reply haskell/adventure.hs]: I ment revealed in "Pieczara" location description, which is already in

Comment (noFrostoo) [reply haskell/adventure.hs]: this is the case now

**Comment by tejotemku** [comment]
Aprove

**Pull request merged**